Update Style+Terminology Guidelines (#693)
kyma-project/community: docs/guidelines/content-guidelines/02-style-and-terminology.md

title: Style and terminology
---

> When you're writing Kyma documentation, refer to the following guidelines for grammar, capitalization, and preferred word choices. These guidelines help that all contributors write in the same way to ensure a uniform flow throughout the whole Kyma documentation.

Author: Good points, thank you!

Use active voice whenever possible. Active voice is clear, concise, and it avoids misinterpretation. It is also easier for non-native speakers to understand. Passive voice is indirect, uses more words, and can be misleading because it reverses the logical order of events.

✅ The endpoint path includes your service name.  
⛔️ Your service name is to be included in the endpoint path.

### Voice and tone

There are different tones for different types of technical documentation. The documentation can range from instructional to somewhat conversational, but always with the goal of helping users understand how to use the product for practical purposes and, in blogs and release notes, also helping business users understand changes.
While writing Kyma documentation, use semi-formal style and imperative mood. The imperative mood tells the reader directly to do something. Use the imperative mood to write instructional documentation such as procedures and tutorials. Other moods can imply optional behavior.

> **NOTE:** Avoid click-level instructions. Avoid using unnecessary words such as "please" or "remember."

✅ Click **Add**.  

> ⛔️ Click the **Add** button.  

Reviewer: > ⛔️ Click the **Add** button.

Why are we introducing this change? 
Although I'm all for brevity and conciseness, I do believe sometimes the "fuller" version sounds better, especially if using just the name might be ambiguous or confusing, or if there are more UI elements with the same or similar text, or the text from the button is repeated in the sentence.

I'd strongly suggest changing this into a suggestion/recommendation rather than a strict rule and leaving the decision to the writer, who knows the context best.

Author: I agree, if the UI is ambiguous we need to be more specific. But how often does that happen?
99% of the time, people write “Click the **Add** button.” when “Add your XY…” would be perfectly sufficient.

Click-level instructions add dependency on the UI strings, creating maintenance effort to make sure that the UI still says “Add” and didn’t change in the meantime.

Author: How about "Avoid click-level instructions unless the UI is confusing. Readers of Kyma documentation are assumed to be fairly tech-savvy."

Reviewer: As a user, I liker (and prefer) click-level instructions. I don't think it's now always about being tech savvy. Imagine if you went to an Azure or AWS dashboard and had no click-level instruction. I'd be lost there, even though the individual controls are self-explanatory. 
And anyway, it's usually the case that the TW writing a document is in charge of maintaining that, so I'd leave it up to the TW. 
If you strongly insist, I'd be OK with adding it as a tip for the undecided that you can decide to follow or not.

Author: I don't think we should ever create Azure/AWS instructions, as long as we don't change company? 
If it's our own Kyma software, we hopefully can improve the usability so people don't get lost.

I'm not opposed to documenting navigation paths etc so that users find the correct screen. And of course, instructions must contain every step that's necessary to finish a task successfully. But the level of detail would be different for tech-savvy Kyma users than for, say, the target group of senior phones. 

The [SAP Guide](https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/80cf1399d10b47b482e833b6dbe9211c.html?state=DRAFT&locale=en-US) makes a good point about not stating the obvious; and frankly I think it makes our life easier if the Kyma and SAP style guides don't differ to much.

Reviewer: > I don't think we should ever create Azure/AWS instructions, as long as we don't change company?
We use external software for development purposes and so sometimes we create documentation that touches that, for example, a guide on how to get logs from the above. 

We hardly ever describe the UI anyway, we do it in the case of Get Started guides, which are targeted for newbies and not only developers, and so being detailed here is justified + it's not like it's huge maintenance if it's just one place. 

Either way, clarity for the user takes priority over our convenience and maintenance load for me. 

I'm still for leaving it up to the author. I think it's a good compromise that we mention it and recommend it but not require it.

Author: Well, then especially if it's someone else's GUI, I would not want to provide click-level instructions. It's sad if the Azure/AWS docs aren't good and/or don't provide the instructions that we need, but I'm opposed to closing that gap in Kyma documentation.

Apart from the occasional "from/in the left navigation,", I think the level of detail in the GSG is absolutely fine. Any less detail would make the instructions useless. 

How about "> **TIP:** If possible, avoid..." ? It's softer than "Avoid...!" but not as soft as "Consider avoiding..."

Reviewer 2: IMO, we can go with the TIP version, but being responsible for our UI, I must say that the click-level instructions make my life easier, and I believe it is the same for the users. So, tbh I'd stick to the current approach.

Author: Then I'd like to go with the "TIP: If possible..." version. Personally, I'm not planning to approve click-level instructions if they can be phrased more abstractly without being less helpful. 

But maybe we're meaning slightly different things here?

Reviewer 2: For me TIP is fine. Just to make sure we're on the same page, by "click-level" I mean e.g. **Click the Add Button** :)

Use active voice whenever possible. Active voice is clear, concise, and it avoids misinterpretation. It is also easier for non-native speakers to understand. Passive voice is indirect, uses more words, and can be misleading because it reverses the logical order of events.

✅ The endpoint path includes your service name.  
⛔️ Your service name is to be included in the endpoint path.

### Voice and tone

There are different tones for different types of technical documentation, which can range from instructional to somewhat conversational. The goal is always to support people using the product and, in blogs and release notes, also to help business users understand changes.

While writing Kyma documentation, use semi-formal style and imperative mood. The imperative mood tells the reader directly to do something. Use the imperative mood to write instructional documentation such as procedures and tutorials. Other moods can imply optional behavior.

> Avoid click-level instructions unless the UI is confusing. Readers of Kyma documentation are typically fairly tech-savvy.  

Author: ```suggestion
> **TIP:** If possible, avoid click-level instructions. Readers of Kyma documentation are typically fairly tech-savvy.  
```

Use camel case for Kubernetes resources and custom resources.

✅ API Gateway is a Kubernetes controller, which operates on APIRule custom resources.  
⛔️ API Gateway is a Kubernetes controller, which operates on API Rule custom resources.

For plurals, add "s".
For titles and navigation, add blank spaces, so that it's natural language instead of camel case.

✅ ConfigMap, ConfigMaps (resource)  
✅ Config Map, Config Maps (resource in titles and navigation)  
⛔️ configmap  
⛔️ config map
  
If you refer to the code specifically, format it as code (more details about [code font](03-formatting.md#code-font-and-bold-font)).

✅ `APIRule`
  
If the words are not used in relation to Kubernetes resources, do not capitalize them.

See the following examples of Kubernetes resources:

* ConfigMap
* CronJob
* CustomResourceDefinition (CRD) - note that "custom resource" alone isn't Kubernetes-specific, thus it's lowercase.
* Deployment
* Function
* Ingress
* Namespace
* Node
* PodPreset
* Pod
* ProwJob
* Secret
* Service
* ServiceBinding
* ServiceClass

> * ServiceInstance

Author: Much better, thank you!

| (you) can                | it is possible to, allows you to, there is the possibility to, you have the option to, … | If there’s no other option, drop the “can”, simply use [imperative](#voice-and-tone). |
| that is                  | i.e.                            | If you must explain a statement with "i.e." or "that is,…", often the first statement adds little value and can be dropped completely, keeping just the part after "i.e.". |
| for example, such as     | e.g.                            | In the middle of a sentence, "such as" is better than "for example". ⛔️ Kyma has many components, for example, Observability, that improve your life. |
| cloud-native (adjective) | cloud native                    | Rule of thumb: Noun compounds usually don't need a hyphen, adjective compounds often do. |
| application              | app                             | Use "Application" to describe an external solution connected to Kyma through Application Connector, "application" to describe a microservice deployed on Kyma or other kind of software. |
| the following         | below, this, the described, ...    | or "as shown in the example"      |

> | the previous, earlier | above, this, the described, ...    | or "as shown in the example"      |

Author: Let's try it out :) I need to see it rendered, not sure whether it'll overwhelmingly bold or just right.

Use active voice whenever possible. Active voice is clear, concise, and it avoids misinterpretation. It is also easier for non-native speakers to understand. Passive voice is indirect, uses more words, and can be misleading because it reverses the logical order of events.

✅ The endpoint path includes your service name.  
⛔️ Your service name is to be included in the endpoint path.

### Voice and tone

There are different tones for different types of technical documentation, which can range from instructional to somewhat conversational. The goal is always to support people using the product and, in blogs and release notes, also to help business users understand changes.

While writing Kyma documentation, use semi-formal style and imperative mood. The imperative mood tells the reader directly to do something. Use the imperative mood to write instructional documentation such as procedures and tutorials. Other moods can imply optional behavior.

> > **TIP:** If possible, avoid click-level instructions. Readers of Kyma documentation are typically fairly tech-savvy.  

Author: ```suggestion
> **TIP:** Consider avoiding click-level instructions. Readers of Kyma documentation are typically fairly tech-savvy.  
```

It's okay to use contractions in the documentation from time to time. However, do not overuse it. Otherwise, your text will become messy and hard to read.

✅ It's okay to use contractions.  
✅ It is okay to not use them as well.

### Articles

Always verify the use of the articles "a", "an", and "the" where appropriate. Use "the" when you refer to a specific example. Use "a" when you refer to something non-specific or hypothetical.

> **NOTE:** Whenever you refer to the name of our product or one of our components, don't use an article:  
  ✅ Kyma is awesome.  
  ⛔️ The Kyma is not awesome.
  ✅ Application Connector  
  ⛔️ the Application Connector

### Punctuation

> Use colons and semicolons sparingly. Use the colon ( : ) to introduce a list of things. The semicolon ( ; ) separates two distinct clauses within one sentence. The phrase after a semicolon is a complete sentence. However, the preferred method is without a colon or semicolon.

Reviewer: While mentioning colons, maybe we could also specify the preferred capitalization after the colon? I remember we discussed it once, whether we should capitalize what comes after the colon, or go with lowercase.

Author: I suggest following the [SAP style guide](https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/e2bfb395649b410ba05a213d80092de5.html?locale=en-US&state=DRAFT)?

Reviewer: Honestly, I don't mind either. But I wonder if we should not stick to what we already discussed and have in the docs, for example [here](https://kyma-project.io/blog/2021/12/7/release-notes-20/)...

Here is the preferred terminology to use in the Kyma documentation:

> * Use American English spelling, not British English.

Reviewer: ```suggestion
* Use the American English spelling, not British English.
```

In such a case, explain the context in the document.

## Capitalization

### Sentence case

Sentence case is the normal style in English: Lowercase for all words, first letter of the first word in a sentence is uppercase. Exception: Uppercase for the first letters of proper nouns and names.

Use sentence case for normal, everyday words in body text. Don't capitalize words just because they are "special" or "important".
Also, we use use sentence case for headings (more details about [headings](03-formatting.md#headings)).

### Upper case

> Upper case means the first letter of a word is capitalized.

Reviewer: This is actually not what Upper case means. If you talk about a letter, upper case means that a letter is --simply -- in upper case (`A`, not `a`). If you talk about a piece of text being written in upper case, it means that THE WHOLE TEXT IS WRITTEN IN UPPER CASE JUST LIKE HERE. 
What you mean is probably **Title Case**, where the first letter of each important word is capitalized ("important" meaning that function words, such as 'of', 'and', etc. are excluded from this and written in lowercase).

See the [Cambridge dictionary](https://dictionary.cambridge.org/dictionary/english/upper-case) for reference.

Author: Thank you, I'll fix that!

Use upper case for Kyma components, such as Application Connector or API Gateway Controller.
Whenever you point to the outside sources, research whether the name of the source starts with a capital letter or not.

### Camel case

Camel case means writing words or phrases without a blank space, using a capitalized letter to separate the words. It's very common in Kubernetes terminology.

Use Camel case for Kubernetes resources and custom resources. For plurals, add "s".
For navigation and titles, use natural language (upper case and blank space).

✅ API Gateway is a Kubernetes controller, which operates on APIRule custom resources.  
⛔️ API Gateway is a Kubernetes controller, which operates on API Rule custom resources.
  
If you refer to the code specifically, format it as code, for example, `APIRule` (more details about [code font](03-formatting.md#code-font-and-bold-font)).
  
However, if the words are not used in relation to Kubernetes resources, do not capitalize them.

See the following examples:

* ConfigMap
* CronJob

> * CustomResourceDefinition (CRD) - note that "custom resource" alone isn't Kubernetes-specific, thus it's lower case.

Reviewer: ```suggestion
* CustomResourceDefinition (CRD) - note that "custom resource" alone isn't Kubernetes-specific, thus it's lowercase.
```
In AmE, it's usually written as one word.
See the [Cambridge dictionary](https://dictionary.cambridge.org/dictionary/english/lowercase) for reference.

Author: Thank you for spotting this!
